python/pfs/drp/stella/fitDistortedDetectorMap.py

```diff
@@ -738,8 +752,18 @@ def fitDistortion(self, bbox, lines, weights, dispersion, seed=0, fitStatic=True
                         newUsed[choose] &= ((np.abs((result.xResid[choose] - dx)/xErr[choose]) < rejection) &
                                             (np.abs((result.yResid[choose] - dy)/yErr[choose]) < rejection))
                 else:
-                    newUsed &= ((np.abs(result.xResid/xErr) < rejection) &
-                                (np.abs(result.yResid/yErr) < rejection))
+                    xResid = np.abs(result.xResid/xErr)
+                    yResid = np.abs(result.yResid/yErr)
+                    keep = newUsed & (xResid < rejection) & (yResid < rejection)
+                    minKeepFrac = 1.0 - self.config.maxRejectionFrac
+                    if keep.sum() < minKeepFrac*used.sum():
+                        xResidLimit = np.percentile(xResid[used], minKeepFrac*100)
+                        yResidLimit = np.percentile(yResid[used], minKeepFrac*100)
+                        keep = newUsed & (xResid < xResidLimit) & (yResid < yResidLimit)
+                        self.log.debug("Standard rejection limit (%f) too severe; using %f, %f",
+                                       rejection, xResidLimit, yResidLimit)
```

Comment:
Is it worth refactoring this and providing a function for computing the rejection limit for both this and lines 513..523 above?